src/runtime/test/render-vdom.spec.tsx

```diff
@@ -902,15 +902,17 @@ describe('render-vdom', () => {
         html: `<svg-attr></svg-attr>`,
       });
 
-      const rect = root.querySelector('rect');
+      let rect = root.querySelector('rect');
       expect(rect.getAttribute('transform')).toBe(null);
 
       root.isOpen = true;
       await waitForChanges();
+      rect = root.querySelector('rect');
       expect(rect.getAttribute('transform')).toBe('rotate(45 27 27)');
 
       root.isOpen = false;
       await waitForChanges();
+      rect = root.querySelector('rect');
```

Comment:
a small difference in how the elements get matched up means that where before a new `rect` element wouldn't be created on re-render it now will be, so the reference to the `rect` element created on the first render isn't to the updated element created on the second re-render

Reply:
It's possible if a stencil user is writing very detailed tests that assert that within a particular component a given HTML element is either destroyed and recreated _or_ that it's conserved across re-renders they might need to rewrite some tests. I have tested in Framework and all the tests there pass with no issues, so that's a good sign at least. Possibly we want to alert users to this? I'm not sure I would really want to classify it as a breaking change though since I think that we are fully within our rights to 'claim' the specific behavior of how Stencil's vdom produces the actual DOM that the user declaratively specifies as an implementation detail, and not itself part of the public API for building Stencil components and so on. I also haven't been able to produce counter-examples for myself which would show different user-facing behavior with this change, only 'under the hood' differences, and I think it's likely that most users wouldn't need to edit their tests with this change